src/Task/MapRoute.pm

```diff
@@ -271,8 +271,25 @@ sub iterate {
 
 	} elsif ( $portals_lut{"$self->{mapSolution}[0]{map} $self->{mapSolution}[0]{pos}{x} $self->{mapSolution}[0]{pos}{y}"}{source} ) {
 		# This is a portal solution
+		
+		
+		if ( distance($self->{actor}{pos_to}, $self->{mapSolution}[0]{pos}) == 0 ) {
+				return unless (timeOut($timeout{ai_portal_wait}));
+				if ($config{route_tryToGuessWrongPortalByDistance}) {
+					my $max_guess_distance = ($config{route_GuessWrongPortalMaxDistance} || 5);
+					for my $portal (@$portalsList) {
+						next unless (distance($self->{actor}{pos_to}, $portal->{pos}) <= $max_guess_distance);
```

Comment:
I suggest picking the closest portal, rather than the one we happen to notice first.

```
my $pos = calcPosition($self->{actor});
for my $portal (sort { distance($pos, $a->{pos}) <=> distance($pos, $b->{pos}) } @$portalsList) {
    last if distance($pos, $portal->{pos}) > $max_guess_distance;
    $self->{actor}->sendMove(@{$portal->{pos}}{qw(x y)});
    last;
}
```

We probably also need to rewrite the route, or the other route code will try to move us back to the missing portal location next cycle.